ThunkLibs/include/common/Guest.h

```diff
@@ -121,6 +122,14 @@ inline Result CallHostFunction(Args... args) {
   // This magic incantation also works in that instance so this is about the best we can do without adding a new attribute to clang for modifying the
   // ABI.
   asm volatile("" : "=r" (host_addr));
+#else
+  // mm0 register is chosen as the register that contains the host_addr.
+  // This register is chosen as it doesn't conflict with vectorcall.
+  // This register also overlaps the x87 st(0) register which is used as float value return.
```

Comment:
What does this mean in practice? Do function pointers returning a `float` not work?

Reply:
In practice this works because we only care about mm0 coming in to the function, on return mm0/st(0) will contain the resulting float/double, overwriting whatever previously was there, which we don't care about at that point.

Reply:
How does the sentence after this (`st(0) is required to be empty coming in to the function because of this`) factor in here? Is that a precondition we can guarantee or do we rely on the guest application to be well-behaving in this regard?

Reply:
If the application is badly behaving here then we will overwrite what is in st(0) by the thunk instruction storing in to mm0, clobbering the data. Which for 32-bit applications you're sort of expecting libraries like GL/Vulkan to stomp on any x87 data anyway. Nothing will survive if it lived in the stack.

Reply:
Alright, makes sense. May I suggest this rephrase?
```
// Use mm0 to pass in host_addr (chosen to avoid conflicts with vectorcall).
// Note this register overlaps the x87 st(0) register (used to return float values),
// so applications that expect this register to be preserved could run into problems.
```
Contains the same information AFAICT but makes it a bit clearer why these things are mentioned in the first place.

Reply:
Updated comment.